activation/handler.go

```diff
@@ -148,9 +148,6 @@ func (h *Handler) processVerifiedATX(
 }
 
 func (h *Handler) SyntacticallyValidate(ctx context.Context, atx *types.ActivationTx) error {
```

Comment:
This function should not receive a `types.ActivationTx` but the respective wire type.

Reply:
Yes, see https://github.com/spacemeshos/go-spacemesh/pull/5789#discussion_r1555355684

Reply:
But the fact that this still uses `types.ActivationTx` and not the wire types is the reason why this PR is so big 😉  see https://github.com/spacemeshos/go-spacemesh/pull/5789#discussion_r1555362673